src/logging.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <google/protobuf/stubs/common.h>
+
+#if GOOGLE_PROTOBUF_VERSION < 3021012
+#define USE_ABSL_LOGGING 0
+#else
+#define USE_ABSL_LOGGING 1
+#endif
+
+#include <grpc/support/log.h>
+
+#if USE_ABSL_LOGGING
+#include <absl/log/globals.h>
+#include <absl/log/initialize.h>
+#include <absl/log/log_sink_registry.h>
+#else
+#include <google/protobuf/stubs/logging.h>
+#endif
+
+#include "ngx_str.hpp"
+
+#if USE_ABSL_LOGGING
+
+class AbslLogSink : public absl::LogSink {
+public:
+    AbslLogSink() { absl::AddLogSink(this); }
+    ~AbslLogSink() override { absl::RemoveLogSink(this); }
```

Comment:
Hmm, It might make sense to reset grpc/legacy protobuf handlers to `NULL` as well.

Reply:
See [ccb30e1](https://github.com/nginxinc/nginx-otel/pull/34/commits/ccb30e11659828c2fb9efa0229cecee0cdcb4237).